tox_poetry_installer/hooks.py

```diff
@@ -196,6 +202,20 @@ def tox_testenv_install_deps(venv: ToxVirtualEnv, action: ToxAction) -> Optional
             dev_deps = []
             logger.info("Env does not install development dependencies, skipping")
 
+        if venv.envconfig.install_group_deps:
+            group_deps = utilities.dedupe_packages(
+                *[
+                    utilities.find_group_deps(group, packages, virtualenv, poetry)
+                    for group in venv.envconfig.install_group_deps
+                ]
+            )
+            logger.info(
+                f"Identified {len(group_deps)} group dependencies to install to env"
+            )
+        else:
+            group_deps = []
+            logger.info("Env does not install group dependencies, skipping")
+
```

Comment:
I think this logic can be simplified. Since the group deps option is a list, we can just iterate over it and if it's empty then we'll get zero dependencies in the result:
```python
group_deps = utilities.dedupe_packages(
    *[
        utilities.find_group_deps(group, packages, virtualenv, poetry)
        for group in venv.envconfig.install_group_deps
     ]
)
logger.info(f"Identified {len(group_deps)} group dependencies to install to env")
```

Reply:
The only consideration here would be the extra logging that is done to make it clear that the option was not set at all, in contrast to the option being set but the specified groups being empty. I'm not sure if the distinction between the two cases is worth it. I personally prefer your suggestion because the less code the better :smile: